ipatests/test_integration/test_commands.py

```diff
@@ -134,6 +135,75 @@ def test_change_sysaccount_password_issue7561(self):
 
+    def test_ldapmodify_password_issue7601(self):
+        user = 'ipauser'
+        original_passwd = 'Secret123'
+        new_passwd = 'userPasswd123'
+        master = self.master
+        base_dn = str(master.domain.basedn)  # pylint: disable=no-member
+
+        # Create a user with a password
+        add_password_stdin_text = "{pwd}\n{pwd}".format(pwd=original_passwd)
+        master.run_command(['ipa', 'user-add', user,
+                            '--first', user,
+                            '--last', user,
+                            '--password'],
+                           stdin_text=add_password_stdin_text)
+        # kinit as that user in order to modify the pwd
+        user_kinit_stdin_text = "{old}\n%{new}\n%{new}\n".format(
+            old=original_passwd,
+            new=original_passwd)
+        master.run_command(['kinit', user], stdin_text=user_kinit_stdin_text)
+        # Retrieve krblastpwdchange and krbpasswordexpiration
+        search_cmd = [
+            'ldapsearch', '-x',
+            '-D', 'cn=directory manager',
+            '-w', master.config.dirman_password,
+            '-b', 'uid={user},cn=users,cn=accounts,{base_dn}'.format(
+                user=user, base_dn=base_dn),
+            '-o', 'ldif-wrap=no',
+            '-LLL',
+            'krblastpwdchange',
+            'krbpasswordexpiration']
+        output = master.run_command(search_cmd).stdout_text.lower()
+
+        # extract krblastpwdchange and krbpasswordexpiration
+        krbchg_pattern = 'krblastpwdchange: (.+)\n'
+        krbexp_pattern = 'krbpasswordexpiration: (.+)\n'
+        krblastpwdchange = re.findall(krbchg_pattern, output)[0]
+        krbexp = re.findall(krbexp_pattern, output)[0]
+
+        # perform ldapmodify on userpassword as dir mgr
+        mod = NamedTemporaryFile()
+        ldif_file = mod.name
+        entry_ldif = textwrap.dedent("""
+            dn: uid={user},cn=users,cn=accounts,{base_dn}
+            changetype: modify
+            replace: userpassword
+            userpassword: {new_passwd}
+        """).format(
+            user=user,
+            base_dn=base_dn,
+            new_passwd=new_passwd)
+        master.put_file_contents(ldif_file, entry_ldif)
+        arg = ['ldapmodify',
+               '-h', master.hostname,
+               '-p', '389', '-D',
+               str(master.config.dirman_dn),   # pylint: disable=no-member
+               '-w', master.config.dirman_password,
+               '-f', ldif_file]
+        master.run_command(arg)
+
+        # Retrieve krblastpwdchange and krbpasswordexpiration
+        output = master.run_command(search_cmd).stdout_text.lower()
+        # extract krblastpwdchange and krbpasswordexpiration
+        newkrblastpwdchange = re.findall(krbchg_pattern, output)[0]
+        newkrbexp = re.findall(krbexp_pattern, output)[0]
+
+        # both should have changed
+        assert newkrblastpwdchange != krblastpwdchange
+        assert newkrbexp != krbexp
+
```

Comment:
The patch looks good. Would it be possible to run an additional similar test (create a user + update userpasswd + retrieve lastpwdchange+expiration) but using ldappasswd (instead of ldapmodify) and check values have changed.
Basically we want DM to be able to user ldapmodify or ldapmodify and get the changes of lastpwdchange+expiration

Reply:
@tbordaz 
thank you for your comments. I updated the test with your suggestion, and also added a sleep before modification of the password (the krb* attributes have at most 1sec precision).